### PR TITLE
CEPH-83573618 - verify Snap mirroring on journaling based image after disable journaling on image

### PR DESCRIPTION
**Pull request by rkhambad**
This PR is to verify the Snapshot based mirroring on image after disabling the Journal mirroring.
Polarion ID : CEPH-83573618 
Polarion Link : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83573618

```
Test case flow:
    1. Create image on cluster-1.
    2. Enable journal mirroring for created image on cluster-1 and run IOs.
    3. Ensure mirror image created on cluster-2.
    4. Disable journal mirroring on image.
    5. enable snapshot mirroring on same image.
    6. Verify for snapshot mirroring and run IOs
    7. cleanup
```
    
Success log : 

RHCS 6.0 - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-75fom/Mirroring_from_journal_to_snapshot_0.log 

RHCS 5.3 - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-2qgmi/Mirroring_from_journal_to_snapshot_0.log

### CODE REVIEW

**Comment by rkhambad** [comment]
latest logs : 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ga5e7/

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    7. cleanup
+    """
+    try:
+        log.info("Starting RBD mirroring test case")
+        config = kw.get("config")
+        mirror1, mirror2 = [
+            rbdmirror.RbdMirror(cluster, config)
+            for cluster in kw.get("ceph_cluster_dict").values()
+        ]
+        poolname = mirror1.random_string() + "_tier_2_rbd_mirror_pool"
```
Comment (sunilangadi2) [inline tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:34]: Are we covering both EC pool and replicated pool for this journal to snapshot mirror conversion?

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        )
+
+        # Disabling journal mirror on image
+        mirror1.disable_mirroring("image", imagespec)
+
+        # enable snapshot mirroring on image
+        mirror1.enable_mirror_image(poolname, imagename, "snapshot")
+        mirror2.wait_for_status(poolname=poolname, images_pattern=1)
+
+        # Verification of mirrored image on cluster
```
Comment (sunilangadi2) [inline tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:57]: what is the unique verification we are doing before it's journal mirrored now it's snapshot based, 
can we verify it through mirror type itself?

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py (hunk 1 above)
Comment (rkhambad) [reply tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:34]: right now we are not covering ec pool for rbd_mirror. As discussed in peer-review meeting.

File: tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py (hunk 2 above)
Comment (rkhambad) [reply tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:57]: Done.

File: suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml [hunk 3]
```diff
@@ -233,3 +233,14 @@ tests:
...
       desc: Testing mirroring of cloned image
+
+  - test:
+      name: Mirroring from journal to snapshot
+      module: test_rbd_mirror_journal_to_snap.py
+      clusters:
+        ceph-rbd1:
+          config:
+            imagesize: 2G
+            io-total: 200M
```
Comment (VasishtaShastry) [inline suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml:244]: Idon't see this module being reused for other purpose than this image and these image size and io-total being used same values and overriden in module
I think these configs can be removed and focused on handling with defaults in testcase module.

File: suites/pacific/rbd/tier-2_rbd_mirror_regression.yaml [hunk 4]
```diff
@@ -239,3 +239,14 @@ tests:
...
       desc: Testing mirroring of cloned image
+
+  - test:
+      name: Mirroring from journal to snapshot
+      module: test_rbd_mirror_journal_to_snap.py
+      clusters:
+        ceph-rbd1:
+          config:
+            imagesize: 2G
+            io-total: 200M
```
Comment (VasishtaShastry) [inline suites/pacific/rbd/tier-2_rbd_mirror_regression.yaml:250]: I don't see this module being reused for other purpose than this image and these image size and io-total being used same values and overriden in module
I think these configs can be removed and focused on handling with defaults in testcase module.

**Review by VasishtaShastry — DISMISSED** [review]
Based on the discussion in peer review meeting, ec pool wrappers are not yet written for mirror utils and will be done later.

mirroring test on replicated pool looks good, moving to approved state with optional comments.
If possible please edit CEPH-83573618 and add a step saying configure mirroring b/w the clusters.

**Comment by rkhambad** [comment]
> Based on the discussion in peer review meeting, ec pool wrappers are not yet written for mirror utils and will be done later.
> 
> mirroring test on replicated pool looks good, moving to approved state with optional comments. If possible please edit CEPH-83573618 and add a step saying configure mirroring b/w the clusters.

Added the step for mirror configuring.

File: suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml [hunk 5]
```diff
@@ -143,93 +143,94 @@ tests:
...
-      name: test-image-delete-from-primary-site
-      module: test-image-delete-primary-site.py
-      clusters:
-        ceph-rbd1:
-          config:
-            imagesize: 2G
-            io-total: 200M
-      polarion-id: CEPH-9501
-      desc: Verify that image deleted at primary site updated at secondary
+  # - test:
```
Comment (sunilangadi2) [inline suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml:146]: why all these tests got commented out?

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml (hunk 3 above)
Comment (rkhambad) [reply suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml]: Done.

File: suites/pacific/rbd/tier-2_rbd_mirror_regression.yaml (hunk 4 above)
Comment (rkhambad) [reply suites/pacific/rbd/tier-2_rbd_mirror_regression.yaml]: Done.

File: suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml (hunk 5 above)
Comment (rkhambad) [reply suites/quincy/rbd/tier-2_rbd_mirror_regression.yaml]: sorry i was testing one change to only this particular test case ....un-commented all .

**Review by sunilangadi2 — APPROVED** [review]
chnages look good

**Review by VasishtaShastry — APPROVED** [review]
lgtm

**Review by manasagowri — APPROVED** [review]
Team review done.

**Pull request merged**